src/main/java/org/takes/http/BkBasic.java

```diff
@@ -73,16 +86,28 @@ public BkBasic(final Take tks) {
     @Override
     public void accept(final Socket socket) throws IOException {
         final InputStream input = socket.getInputStream();
+        final RqLive req = new RqLive(input);
+        boolean keep = false;
+        final Iterator<String> values = new RqHeaders.Base(req)
+            .header(CONNECTION).iterator();
+        if (values.hasNext()) {
+            do {
+                keep = KEEP_ALIVE.equals(values.next());
```

Comment:
@pinaf  the `if` is not necessary, in the next line there is a `while` with `!keep` condition. This cover the case for multiple headers with the same name. But in the case of a comma separated values, the solution will be given by using `contains()` instead of `equals()`